examples/rust/section01-introduction/Cargo.toml

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "section01-introduction"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+opendal = { version = "0.34.0", features = ["services-webdav", "services-gdrive"] }
```

Comment:
Please use latest version and don't enable not needed feature.

Reply:
when switch to 0.37.0, an error occurs 🤔
```txt
error[E0432]: unresolved import `futures::AsyncRead`
  --> /home/flash/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/opendal-0.37.0/src/types/reader.rs:25:5
   |
25 | use futures::AsyncRead;
   |     ^^^^^^^^^^^^^^^^^^ no `AsyncRead` in the root
   |
help: consider importing this trait instead
   |
25 | use tokio::io::AsyncRead;
```

Reply:
I'm guessing you need to add `futures` in `Cargo.toml`?

Reply:
```
#[cfg(feature = "std")]
#[doc(no_inline)]
pub use futures_io::{AsyncBufRead, AsyncRead, AsyncSeek, AsyncWrite};
#[cfg(feature = "std")]
#[doc(no_inline)]
pub use futures_util::{AsyncBufReadExt, AsyncReadExt, AsyncSeekExt, AsyncWriteExt};
```

`AsyncRead` is only exported under `std`.

Reply:
Please add `futures` in `Cargo.toml` (we will remove it in the future)

Reply:
should we add `examples/rust/section01-introduction` to the root `Cargo.toml` workspace.members ?
If no, when we run `cargo run` in `section01-introduction`, it will occurs the following error
```txt
error: current package believes it's in a workspace when it's not:
current:   /home/flash/projects/opendal/examples/rust/section01-introduction/Cargo.toml
workspace: /home/flash/projects/opendal/Cargo.toml

this may be fixable by adding `examples/rust/section01-introduction` to the `workspace.members` array of the manifest located at: /home/flash/projects/opendal/Cargo.toml
Alternatively, to keep it out of the workspace, add the package to the `workspace.exclude` array, or add an empty `[workspace]` table to the package's manifest.
```

Reply:
> should we add `examples/rust/section01-introduction` to the root `Cargo.toml` workspace.members ?

No.



> If no, when we run `cargo run` in `section01-introduction`, it will occurs the following error

We can add an empty `[workspace]` in `Cargo.toml`

Reply:
For example: 

![image](https://github.com/apache/incubator-opendal/assets/5351546/0d8e092a-517e-4ce8-aa02-0ecb6ab5943d)